clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -7668,6 +7669,47 @@ struct LoopIterationSpace final {
   Expr *FinalCondition = nullptr;
 };
 
+class ForSubExprChecker : public RecursiveASTVisitor<ForSubExprChecker> {
+  const llvm::SmallSet<Decl *, 4> *CollapsedLoopVarDecls;
+  VarDecl *ForbiddenVar;
+  SourceRange ErrLoc;
+
+public:
+  explicit ForSubExprChecker(
+      const llvm::SmallSet<Decl *, 4> *CollapsedLoopVarDecls)
```

Comment:
1. Pass by reference
2. Use lvm::SmallSetImpl<Decl*>

Reply:
There isn't a SmallSetImpl, so I've changed all these to SmallPtrSet{,Impl} instead.